lib/utils/get-address.js

```diff
@@ -1,15 +1,57 @@
-// return first external ipv4 address
-var os = require("os");
+// return specific address on specific NIC, or first external ipv4 address on on specific NIC or eth0 (default)
+// os and env arguments make the code testable
+module.exports = function(logger, env, os) {
+  logger = logger || console;
+  os = os || require("os");
+  env = env || process.env;
+  return function(interfaces) {
+    let networkInterfaceId = env["NETWORK_INTERFACE_ID"] || "eth0";
+    interfaces = interfaces || os.networkInterfaces();
+    if (!interfaces[networkInterfaceId]) {
+      return getFirstAvailableAddress(networkInterfaceId, logger, os);
+    }
+    let interfaceItemIndex = parseInt(env["NETWORK_INTERFACE"] || 0);
+    if (
+      isNaN(interfaceItemIndex) ||
+      interfaceItemIndex >= interfaces[networkInterfaceId].length
+    ) {
+      return getFirstAvailableAddress(networkInterfaceId, logger, os);
+    }
+    return interfaces[networkInterfaceId][interfaceItemIndex].address;
+  };
+};
 
-const NETWORK_INTERFACE = parseInt(process.env["NETWORK_INTERFACE"]) || 0;
+function getFirstAvailableAddress(interfaceId, logger, os) {
+  const interfaces = os.networkInterfaces();
+  const candidates = Object.keys(interfaces).reduce(
```

Comment:
can we also sort this array to make sure index-based behaviour is consistent?

Reply:
no problem

Reply:
this might be a breaking change, but that's not a major problem